### ISSUE TITLE
Selected item occasionally jumps to first item whilst hovering.

### ISSUE THREAD

**Issue by edwardhorsford** [alphagov/accessible-typeahead#102]
Whilst hovering over menu items, the selected item sometimes briefly jumps to the first (focused) item.

It *might* be related to how we revert visibly selected to the focused item when you mouse out of the menu.

Example:
![location-picker-bug-hover](https://cloud.githubusercontent.com/assets/2204224/25802819/00540dea-33ec-11e7-9d61-efb6e675d082.gif)

**Comment by tvararu** [comment]
Yes, it's related to that and how the mouse handlers are configured.
